### ISSUE TITLE
missing setVideoScalingMode

### ISSUE THREAD

**Issue by stmkjp** [protyposis/ITEC-MediaPlayer#62]
Hi,
thanks for the great library!

I'm trying to implement your mediaplayer library and it's good but i have one problem.
( put video on textureview using mediaplayer )
I'm currently building app using portrait only mode, and it's ok when I load landscape video (scaled and no distortion) - same as default mediaplayer, 
![screenshot_1492074880](https://cloud.githubusercontent.com/assets/5444842/24998511/504b7228-2075-11e7-838c-7cbf1e5d5661.png)

but when I load portrait video, aspect ratio not maintained 
![screenshot_1492074888](https://cloud.githubusercontent.com/assets/5444842/24998587/9816eeb6-2075-11e7-91ce-7143f42f839c.png)

android default mediaplayer has setVideoScalingMode(MediaPlayer.VIDEO_SCALING_MODE_SCALE_TO_FIT_WITH_CROPPING) method

but  I can't find corresponding method in MediaPlayer-Extended library.

Is there any way to resolve this? Any help would be really appreciated!

Best,
JP

**Comment by protyposis** [comment]
Yes, MPX does not implement the `setVideoScalingMode` API method. 

To maintain the aspect ratio, you can extend the `TextureView` and override `onMeasure` like I am doing in the [`VideoView` here](https://github.com/protyposis/MediaPlayer-Extended/blob/5225e50aa7a11b439abae35371fe79dd5dcd080f/MediaPlayer/src/main/java/net/protyposis/android/mediaplayer/VideoView.java#L231), which always resizes the View to fit it's aspect ratio to the video frame. You can then position this view within a layout. 

If you want cropping, then there is currently no other way than to also override `onMeasure` to maintain the aspect ratio and programmatically set the size of an enclosing layout to crop the video as desired. 

Btw. it is recommended to use `SurfaceView` over `TextureView` because the latter draws much more power.

**Comment by devingDev** [comment]
this is stupid your view should handle it. what i now got from your lib is basically the same as the one from android. stretched video and nothing else.
And no I do not plan to dig deeper into this thing just to find out its not possible.

**Comment by protyposis** [comment]
Thank you for this very productive and enlightening comment, and good luck with your future endeavors.
